### ISSUE TITLE
Support custom fields on a mocked GraphQL response

### ISSUE THREAD

**Issue by stefinie123** [kettanaito/msw#1200]
### Scope

Improves an existing behavior

### Compatibility

- [ ] This is a breaking change

### Feature description

I'm using msw for mocking grapghql api calls in react testing library tests of my application.
I have a requirement to pass following object as the response context, to simulate an error response behavior
```
{
            "networkError": {
                "statusCode": "404"
            },
            "message": "Not Found",
            "metadata": {
                "errorCode": "Repository Not Accessible",
                "additionalData": "http://github.com/testUser/repo-to-be-deleted"
            }
        }
```
However when setting the response context it only allows me to set `data`, `headers`, `delay`, `extensions` and some other predefined fields for the context.
I need to be able to set custom values, for an example as per the above object I need to set `metadata` and `message` fields

**Comment by kettanaito** [comment]
Hey, @stefinie123. Thanks for suggesting this!

We've chosen a more restricting API of `ctx.data` and `ctx.errors` to encourage people to adhere to the GraphQL specification. As I understand, it's still up to the contract between the server and the client what response fields are returned. We should support this in MSW in some shape or form. 

What about `ctx.field()` that would set each response field individually?

```js
return res(
  ctx.data({ user: { ... } }),
  ctx.field('metadata', {
    errorCode: 'Repository Not Accessible',
    additionalData: 'https://example.com'
  }),
  ctx.field('customProp', 23)
)
```

This kind of API would make adding custom fields a conscious choice as opposed to using a generic `ctx.json` which would also conflict with the `ctx.data`/`ctx.errors` functionality. It'd also be forbidden to set the keys "data", "errors" or "extensions" when using `ctx.field`.

What are your thoughts around this?

**Comment by kettanaito** [comment]
I would love to have an excerpt from the GraphQL specification around custom fields and error-handling in particular. We need to be sure we're not enabling mock behaviors that go against the official specification.

**Comment by stefinie123** [comment]
Hi @kettanaito , as you have suggested it would be great if you can support `ctx.field()` , so that we can set each custom field individually

**Comment by kettanaito** [comment]
Nice! Would you like to help me with a pull request? You can see ctx.data for inspiration, and I'll help you with the review and tests.

**Comment by SalahAdDin** [comment]
Is it implemented now?

**Comment by kettanaito** [comment]
Hey, @SalahAdDin. If the issue is open then it's not implemented. You're welcome to join and help us implement it: I've shared the details [above](https://github.com/mswjs/msw/issues/1200#issuecomment-1098017792).

**Comment by Poivey** [comment]
Hello @kettanaito. I would like to implement this feature.

Is this specification good enough ? https://github.com/mswjs/msw/issues/1200#issuecomment-1098017792
Or do you think now this could cause an issue ?

**Comment by kettanaito** [comment]
Hey, @Poivey! 👋 Nice to see you. 

The specification above looks okay to me, I don't think it will introduce any issues. We probably want to forbid using "data", "errors" and "extensions" as the first argument to `ctx.field()` on both type and runtime level (can use already present `invariant`). Let me know if you get to working on this, I'd love to collaborate in a pull request.

**Comment by Poivey** [comment]
Ok I will start working on it, I will create a pull request as soon as I have the first implementation to get your feedback !

**Comment by kettanaito** [comment]
## Released: v0.42.1 🎉

This has been released in v0.42.1!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v0.42.1)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/0.42.1)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
